checks/raw/branch_protection_test.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2022 Security Scorecard Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package raw
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/ossf/scorecard/v4/clients"
+)
+
+var branch = "master"
+
+func Test_getBranchName(t *testing.T) {
+	t.Parallel()
+	type args struct {
+		branch *clients.BranchRef
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "simple",
+			args: args{
+				branch: &clients.BranchRef{
+					Name: &branch,
+				},
+			},
+			want: master,
+		},
+		{
+			name: "empty name",
+			args: args{
+				branch: &clients.BranchRef{},
+			},
+			want: "",
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			if got := getBranchName(tt.args.branch); got != tt.want {
+				t.Errorf("getBranchName() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_getBranchMapFrom(t *testing.T) {
+	t.Parallel()
+	type args struct {
+		branches []*clients.BranchRef
+	}
+	//nolint
+	tests := []struct {
+		name string
+		args args
+		want branchMap
+	}{
+		{
+			name: "simple",
+			args: args{
+				branches: []*clients.BranchRef{
+					{
+						Name: &branch,
+					},
+				},
+			},
+			want: branchMap{
+				master: &clients.BranchRef{
+					Name: &branch,
+				},
+			},
+		},
+	}
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			if got := getBranchMapFrom(tt.args.branches); !reflect.DeepEqual(got, tt.want) {
```

Comment:
Instead of `reflect.DeepEqual`, can use `cmp.Equal` (package `google/go-cmp`). It also lets you pretty print the diff between the 2 structs using `cmp.Diff`.